crates/core/tedge_agent/src/restart_manager/actor.rs

```diff
@@ -244,8 +245,13 @@ impl RestartManagerActor {
         let system_config = SystemConfig::try_new(&self.config.config_dir)?;
 
         // Check if sudo command is available
-        match which(SUDO) {
-            Ok(sudo) => {
+        let sudo = which(SUDO).ok();
+        if self.config.is_sudo_enabled && sudo.is_none() {
+            warn!("`sudo.enable` is `true` but sudo wasn't found in $PATH");
+        }
+
+        match sudo {
+            Some(sudo) if self.config.is_sudo_enabled => {
```

Comment:
This pattern combining `which` and `config.is_sudo_enabled` to get an `Option<Path>` itself pattern-matched to run a command with or without `sudo` is used in too many places with the risk to be non consistent (as proven by the need for this PR).

A first step to fix that is to stop passing boolean values telling to use `sudo` or not.

Instead the `restart_manager` (as well as the `software_manager` and also `tedge_writer` and `plugin_sm`) should be given a capability object, a struct enabling the capability to run a command with root privileges (*possibly* using `sudo` under the hood).

Something along these lines:

```rust
pub struct Sudo {
      sudo: Option<PathBuf>
}

impl Sudo {
     pub fn new(config: &TEdgeConfig) -> Sudo { ... }
     pub fn new_command<S: AsRef<OsStr>>(&self program: S) -> Command { ... }
}
```

Where to define this capability object? in `tedge_utils`? `tedge_config`? I would go with the latter.